src/Mremi/UrlShortener/Provider/Bitly/BitlyProvider.php

```diff
@@ -65,15 +65,21 @@ public function shorten(LinkInterface $link, $domain = null)
     {
         $client = $this->createClient();
 
-        $response = $client->get(sprintf('/v3/shorten?access_token=%s&longUrl=%s&domain=%s',
-            $this->auth->getAccessToken(),
-            urlencode($link->getLongUrl()),
-            $domain
-        ), $this->options);
+        $this->options['headers']['Authorization'] = 'Bearer ' . $this->auth->getAccessToken();
 
-        $response = $this->validate($response->getBody()->getContents());
+        $response = $client->post('/v4/shorten', array_merge(
+            array(
+                'json' => array(
+                    'domain' => $domain,
+                    'long_url' => $link->getLongUrl(),
+                ),
+            ),
+            $this->options
+        ));
+
+        $response = $this->validate($response->getBody()->getContents(), $response->getStatusCode());
```

Comment:
The whole response should be passed, I mean:
```suggestion
        $response = $this->validate($response);
```